### PR TITLE
awk, cut, ed, sed: add and refresh pages

### PR DESCRIPTION
**Pull request by EmilySeville7cfg**
- [ ] The page (if new), does not already exist in the repository.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

**Version of the command being documented (if known):**

# ToDo

- [x] GNU sed
- [x] GNU cut
- [x] GNU ed
- [x] GNU awk
- [x] FreeBSD sed
- [x] FreeBSD cut
- [x] FreeBSD ed
- [x] FreeBSD awk

### CODE REVIEW

File: pages/common/awk.md [hunk 1]
```diff
@@ -1,36 +1,36 @@
 # awk
 
 > A versatile programming language for working on files.
-> More information: <https://github.com/onetrueawk/awk>.
+> More information: <https://manned.org/awk.1>.
```
Comment (sbrl) [inline pages/common/awk.md:4]: The project page is preferable to a man page. Let's revert this change and keep it as it was.

File: pages/common/awk.md [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
+- Print the specified field range of each line:
 
-`awk '($10 == value)'`
+`awk "{ for (i=1; i <= 10; i++) print \$i }"`
 
-- Print all the lines which the 10th column value is between a min and a max :
+- Print the version:
 
-`awk '($10 >= min_value && $10 <= max_value)'`
+`awk --version`
```
Comment (sbrl) [inline pages/common/awk.md:36]: On a page such as this, documenting how to show the version of `awk` is not a good use of examples - especially since it follows the conventions of other commands. Is there anything more useful we could do here instead?

File: pages/common/cut.md [hunk 3]
```diff
@@ -1,28 +1,24 @@
 # cut
 
 > Cut out fields from stdin or files.
-> More information: <https://www.gnu.org/software/coreutils/cut>.
+> More information: <https://manned.org/cut.1>.
```
Comment (sbrl) [inline pages/common/cut.md:4]: Again, the GNU coreutils link is preferable to a man page.

File: pages/common/cut.md [hunk 4]
```diff
@@ -1,28 +1,24 @@
...
 
-- Cut out the 2nd and 10th fields of each line, using a semicolon as a delimiter:
+- Print the version:
 
-`cut -d'{{;}}' -f{{2,10}}`
-
-- Cut out the fields 3 through to the end of each line, using a space as a delimiter:
-
-`cut -d'{{ }}' -f{{3-}}`
+`cut --version`
```
Comment (sbrl) [inline pages/common/cut.md:24]: Again, showing the version is not a good use of examples on a page like this.

File: pages/common/ed.md [hunk 5]
```diff
@@ -1,36 +1,36 @@
 # ed
 
 > The original Unix text editor.
-> More information: <https://man.archlinux.org/man/ed.1>.
+> More information: <https://manned.org/ed.1>.
 
-- Start ed, editing an empty document (which can be saved as a new file in the current directory):
+- Start an interactive editor session:
```
Comment (sbrl) [inline pages/common/ed.md:6]: The previous example description seems to be more clear to me.

File: pages/common/ed.md [hunk 6]
```diff
@@ -1,36 +1,36 @@
...
+- Replace a string with the specified replacement for all lines:
 
-`w {{filename}}`
+`,s/{{regular_expression}}/{{replacement}}/g`
 
-- Quit ed:
+- Print the version:
 
-`q`
+`ed --version`
```
Comment (sbrl) [inline pages/common/ed.md:36]: Same again. No point in documenting how to show the version.

File: pages/osx/sed.md [hunk 7]
```diff
@@ -1,36 +1,32 @@
...
 
 > Edit text in a scriptable manner.
-> More information: <https://ss64.com/osx/sed.html>.
+> More information: <https://manned.org/sed.1>.
 
-- Replace the first occurrence of a string in a file, and print the result:
+- Execute an expression:
 
-`sed 's/{{find}}/{{replace}}/' {{filename}}`
+`sed "{{expression}}"`
```
Comment (sbrl) [inline pages/osx/sed.md:8]: I feel a number of the new examples on this page are too generic? I'm not convinced that the new layout of this page is more useful than the old one - though I do recognise the merits of showing the generic syntax first.

**Review by sbrl — COMMENTED** [review]
Thanks for the page edits! I've left some comments below for you to review.

I'm curious about your motives here. These are some *major* changes to the pages some quite core commands. Some of the changes seem good, others less so.

File: pages/common/awk.md (hunk 2 above)
Comment (EmilySeville7cfg) [reply pages/common/awk.md:36]: I suggest to decide when to show examples for `--help`|`--version` options. Maybe don't show them at all and write smth more useful.

File: pages/common/awk.md (hunk 1 above)
Comment (EmilySeville7cfg) [reply pages/common/awk.md:5]: But there is no language description directly available via this suggested url. It may confuse users.

File: pages/common/cut.md (hunk 3 above)
Comment (EmilySeville7cfg) [reply pages/common/cut.md:5]: When we have to use smth instead of manned.org? As a user I like unified manned.org interface. ;)

File: pages/osx/sed.md (hunk 7 above)
Comment (EmilySeville7cfg) [reply pages/osx/sed.md]: I don't know how to escape placeholder character to write AWK code sample for:

```
`awk '{{{ print $1 }}}'`
```

It is useful because it shows difference between sed and awk.

Comment (EmilySeville7cfg) [reply pages/osx/sed.md]: Fixed. ;)

File: pages/common/ed.md (hunk 5 above)
Comment (EmilySeville7cfg) [reply pages/common/ed.md:7]: I've changed description to `Start an interactive editor session with empty document`.

File: pages/common/awk.md (hunk 1 above)
Comment (sbrl) [reply pages/common/awk.md:5]: I'm not sure what you're referring to here. Could you clarify please @EmilySeville7cfg?

File: pages/common/cut.md (hunk 3 above)
Comment (sbrl) [reply pages/common/cut.md]: I don't understand what you're saying here. What's `smth`?

File: pages/common/awk.md [hunk 8]
```diff
@@ -1,36 +1,33 @@
...
-`awk --file {{path/to/file}}`
+`awk --file {{path/to/script.awk}}`
 
 - Check a script for syntax errors:
 
-`awk --lint --file {{path/to/file}}`
+`awk --lint --file {{path/to/script.awk}}`
 
-- Execute an expression with the specified field separator:
+- Execute commands with the specified field separator:
```
Comment (sbrl) [inline pages/common/awk.md:19]: Let's be more specific here.
```suggestion
- Execute an AWK expression with the specified field separator:
```

`expression` is the most technically correct language here.

File: pages/common/cut.md [hunk 9]
```diff
@@ -1,7 +1,8 @@
 # cut
 
 > Cut out fields from stdin or files.
-> More information: <https://manned.org/cut.1>.
+> Get help or version: cut --help|--version.
```
Comment (sbrl) [inline pages/common/cut.md:4]: This line is not necessary.
```suggestion
```

File: pages/common/ed.md [hunk 10]
```diff
@@ -1,21 +1,22 @@
...
-- Start an interactive editor session:
+- Start an interactive editor session with an empty document:
 
 `ed`
 
-- Start an interactive editor session with the specified prompt:
+- Start an interactive editor session with an empty document and the specified prompt:
 
-`ed --prompt='> '`
+`ed --prompt="> "`
```
Comment (sbrl) [inline pages/common/ed.md:13]: Single quotes are preferable here to avoid a `$` for example being interpreted by the shell.

File: pages/common/sed.md [hunk 11]
```diff
@@ -1,36 +1,33 @@
...
+`sed '{{s/apple/mango/g}}'`
 
-- Replace all occurrences of an extended regular expression in a file, and print the result:
+- Execute a script:
 
-`sed -r 's/{{regular_expression}}/{{replace}}/g' {{filename}}`
+`sed --file={{path/to/script.sed}}`
 
-- Replace all occurrences of a string in a file, overwriting the file (i.e. in-place):
+- Execute commands with enabled extended regular expressions:
```
Comment (sbrl) [inline pages/common/sed.md:15]: Perhaps this will read better? Also, `expression` is more technically correct than `command`.
```suggestion
- Execute an expression, enabling extended regular expressions:
```

File: pages/common/sed.md [hunk 12]
```diff
@@ -1,36 +1,33 @@
...
+`sed --regexp-extended '{{s/(apple)/\U\1/g}}'`
 
-- Replace only on lines matching the line pattern:
+- Execute commands without automatic buffer printing:
 
-`sed '/{{line_pattern}}/s/{{find}}/{{replace}}/' {{filename}}`
+`sed --quiet '{{1p}}'`
 
-- Delete lines matching the line pattern:
+- Execute commands and replace file with it's output:
```
Comment (sbrl) [inline pages/common/sed.md:23]: Perhaps this will read better? Also, `expression` is more technically correct than `command`.
```suggestion
- Execute an expression and replace a file with it's output:
```

File: pages/osx/cut.md [hunk 13]
```diff
@@ -0,0 +1,21 @@
+# cut
```
Comment (sbrl) [inline pages/osx/cut.md:1]: Same here. Why the copy?

File: pages/osx/ed.md [hunk 14]
```diff
@@ -0,0 +1,25 @@
+# ed
```
Comment (sbrl) [inline pages/osx/ed.md:1]: Same here. Why the copy?

File: pages/osx/sed.md [hunk 15]
```diff
@@ -1,32 +1,33 @@
 # sed
 
 > Edit text in a scriptable manner.
-> More information: <https://manned.org/sed.1>.
+> See also: `awk`, `ed`.
+> More information: <https://manned.org/man/freebsd-13.0/sed.1>.
 
-- Execute an expression:
+- Execute commands:
```
Comment (sbrl) [inline pages/osx/sed.md:7]: This is not correct. `expression` is the more accurate technical language here.
```suggestion
- Execute an expression:
```

File: pages/osx/sed.md [hunk 16]
```diff
@@ -1,32 +1,33 @@
...
-`sed "{{expression}}"`
+`sed '{{s/apple/mango/g}}'`
 
 - Execute a script:
 
-`sed --file={{path/to/file}}`
+`sed -f {{path/to/script.sed}}`
 
-- Execute an expression with enabled extended regular expressions:
+- Execute commands with enabled [E]xtended regular expressions:
```
Comment (sbrl) [inline pages/osx/sed.md:15]: Same here.
```suggestion
- Execute an expression with enabled [E]xtended regular expressions:
```

File: pages/osx/sed.md [hunk 17]
```diff
@@ -1,32 +1,33 @@
...
 
-`sed --regexp-extended "{{expression}}"`
+`sed -E '{{s/(apple)/\U\1/g}}'`
 
-- Execute a script and replace file with it's output:
+- Execute commands without automatic buffer printing:
```
Comment (sbrl) [inline pages/osx/sed.md:19]: Same here.
```suggestion
- Execute am expression without automatic buffer printing:
```

File: pages/osx/sed.md [hunk 18]
```diff
@@ -1,32 +1,33 @@
...
 
-`sed --in-place --file={{path/to/file}}`
+`sed -n '{{1p}}'`
 
-- Execute an expression without automatic buffer printing:
+- Execute commands and replace file with it's output:
```
Comment (sbrl) [inline pages/osx/sed.md:23]: Same here.
```suggestion
- Execute an expression and replace the file with the output:
```

File: pages/osx/awk.md [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+
+- Execute commands:
+
+`awk '{{{ print gensub("apple", "mango", "g") }}}'`
+
+- Execute a script:
+
+`awk -f {{path/to/script.awk}}`
+
+- Execute commands with the specified field separator:
```
Comment (sbrl) [inline pages/osx/awk.md:15]: `expression` is more technically accurate.
```suggestion
- Execute an expression with the specified field separator:
```

File: pages/osx/awk.md [hunk 20]
```diff
@@ -0,0 +1,29 @@
+# awk
+
+> A versatile programming language for working on files.
+> See also: `sed`, `ed`.
+> More information: <https://manned.org/man/freebsd-13.0/awk.1>.
+
+- Execute commands:
```
Comment (sbrl) [inline pages/osx/awk.md:7]: Perhaps this will read better?
```suggestion
- Execute an expression:
```

File: pages/common/sed.md [hunk 21]
```diff
@@ -1,36 +1,33 @@
...
+`sed --file={{path/to/script.sed}}`
 
-- Replace all occurrences of a string in a file, overwriting the file (i.e. in-place):
+- Execute commands with enabled extended regular expressions:
 
-`sed -i 's/{{find}}/{{replace}}/g' {{filename}}`
+`sed --regexp-extended '{{s/(apple)/\U\1/g}}'`
 
-- Replace only on lines matching the line pattern:
+- Execute commands without automatic buffer printing:
```
Comment (sbrl) [inline pages/common/sed.md:19]: `expression` is more technically correct than `command`.
```suggestion
- Execute an expression without automatic buffer printing:
```

File: pages/common/sed.md [hunk 22]
```diff
@@ -1,36 +1,33 @@
 # sed
 
 > Edit text in a scriptable manner.
-> More information: <https://man.archlinux.org/man/sed.1>.
+> See also: `awk`, `ed`.
+> More information: <https://manned.org/sed.1>.
 
-- Replace the first occurrence of a regular expression in each line of a file, and print the result:
+- Execute commands:
```
Comment (sbrl) [inline pages/common/sed.md:7]: `expression` is more technically correct than `command`.
```suggestion
- Execute an expression:
```

File: pages/common/awk.md [hunk 23]
```diff
@@ -1,36 +1,33 @@
...
 
-`awk --assign "{{variable}}={{value}}" "{{expression}}"`
+`awk '{ print gensub("{{regular_expression}}", "{{replacement}}", "g") }'`
 
 - Print the specified field of each line:
 
-`awk "{ print \$1 }"`
+`awk '{ print ${{1}} }'`
 
 - Print the specified field range of each line:
```
Comment (sbrl) [inline pages/common/awk.md:31]: I would say that the `Execute an expression with the specified variables` example would be more useful than this one.

File: pages/common/awk.md [hunk 24]
```diff
@@ -1,36 +1,33 @@
 # awk
 
 > A versatile programming language for working on files.
+> See also: `sed`, `ed`.
 > More information: <https://manned.org/awk.1>.
 
-- Execute an expression:
+- Execute commands:
```
Comment (sbrl) [inline pages/common/awk.md:7]: `expression` is more technically correct than `command`.
```suggestion
- Execute an expression:
```

**Review by sbrl — CHANGES_REQUESTED** [review]
I've left some more comments below for you to review.

**Review by blueskyson — COMMENTED** [review]
I suggest closing this PR then sending one PR per page, so that we can at least approve some pages and leave other pages in discussions.

**Comment by EmilySeville7cfg** [comment]
> I suggest closing this PR then sending one PR per page, so that we can at least approve some pages and leave other pages in discussions.

Nice idea. We can restrict PRs with multiple page edits to some simple cases as fixing placeholders, man page urls and etc, but not big rewrite as this one. PRs must be easy to check. 😁

**Comment by EmilySeville7cfg** [comment]
@blueskyson, I am closing this PR because I transferring changes to the new ones.
